compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/JvmCreateTypeGen.java

```diff
@@ -0,0 +1,1720 @@
+        ClassWriter cw = new BallerinaClassWriter(COMPUTE_FRAMES);
+        cw.visit(V1_8, ACC_PUBLIC + ACC_SUPER, moduleTypesClass, null, OBJECT, null);
+        generateCreateTypesMethod(cw, module.typeDefs, moduleInitClass, symbolTable);
+        cw.visitEnd();
+        byte[] bytes = jvmPackageGen.getBytes(cw, module);
+        jarEntries.put(moduleTypesClass + ".class", bytes);
+    }
+
+    void generateCreateTypesMethod(ClassWriter cw, List<BIRTypeDefinition> typeDefs,
+                                   String moduleInitClass, SymbolTable symbolTable) {
+
+        createTypesInstance(cw, typeDefs, moduleInitClass);
+        List<String> populateTypeFuncNames = populateTypes(cw, typeDefs, moduleInitClass, symbolTable);
+
+        MethodVisitor mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC, CREATE_TYPES_METHOD, "()V", null, null);
+        mv.visitCode();
+
+        // Invoke create-type-instances method
+        mv.visitMethodInsn(INVOKESTATIC, moduleTypesClass, CREATE_TYPE_INSTANCES_METHOD, "()V", false);
+
+        // Invoke the populate-type functions
+        for (String funcName : populateTypeFuncNames) {
+            mv.visitMethodInsn(INVOKESTATIC, moduleTypesClass, funcName, "()V", false);
+        }
+        mv.visitInsn(RETURN);
+        mv.visitMaxs(0, 0);
+        mv.visitEnd();
+    }
+
+    private void createTypesInstance(ClassWriter cw, List<BIRTypeDefinition> typeDefs, String moduleInitClass) {
+        int instanceSplits = createTypesInstanceSplits(cw, typeDefs, moduleInitClass);
+        MethodVisitor mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC, CREATE_TYPE_INSTANCES_METHOD, "()V", null, null);
+        mv.visitCode();
+        for (int i = 0; i < instanceSplits; i++) {
+            mv.visitMethodInsn(INVOKESTATIC, moduleTypesClass, CREATE_TYPE_INSTANCES_METHOD + i, "()V", false);
+        }
+        mv.visitInsn(RETURN);
+        mv.visitMaxs(0, 0);
+        mv.visitEnd();
+    }
+
+    private int createTypesInstanceSplits(ClassWriter cw, List<BIRTypeDefinition> typeDefs, String typeOwnerClass) {
+
+        if (typeDefs.isEmpty()) {
+            return 0;
+        }
+        MethodVisitor mv = null;
+        int bTypesCount = 0;
+        int methodCount = 0;
+
+        // Create the type
+        for (BIRTypeDefinition optionalTypeDef : typeDefs) {
+            String name = optionalTypeDef.internalName.value;
+            BType bType = optionalTypeDef.type;
+            if (bType.tag == TypeTags.RECORD || bType.tag == TypeTags.OBJECT || bType.tag == TypeTags.ERROR ||
+                    bType.tag == TypeTags.UNION || bType.tag == TypeTags.TUPLE) {
+                if (bTypesCount % MAX_TYPES_PER_METHOD == 0) {
+                    mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC, CREATE_TYPE_INSTANCES_METHOD + methodCount++, "()V",
+                            null, null);
+                    mv.visitCode();
+                }
+            } else {
```

Comment:
Shall we inverse this if check? So that for the other types it will return early and we don't need an `else` part.